apps/meteor/app/api/server/v1/moderation.ts

```diff
@@ -243,3 +249,30 @@ API.v1.addRoute(
 		},
 	},
 );
+
+API.v1.addRoute(
+	'moderation.reportUser',
+	{
+		authRequired: true,
+		validateParams: isModerationReportUserPost,
+	},
+	{
+		async post() {
+			const { userId, description } = this.bodyParams;
+
+			const {
+				user: { _id, name, username, createdAt },
+			} = this;
+
+			const reportedUser = await Users.findOneById(userId, { projection: { _id: 1, name: 1, username: 1 } });
```

Comment:
This could be done, but there's a couple of regards.
1. Will we update these fields for every report when the user is updated? If not, it should be correctly described as a field that was only recorded at the time of the report. If we are going to update then we should implement the reactivity.

2. Are we going to use `createdAt` for something? As far as I know there's no use for it currently. This one doesn't need to be updated though, and should be easier to include.

Reply:
Thanks for bringing up the above points; I understand that's the thing to consider.

After thinking about it for a while and considering the previous comment, _we might have to consider modifying the design_. 

**Approach 1:**
I think this is more suitable for user reports:
```ts
{
_id: string,
reportedUser: IUser<roles[], _id, name, uname, createdAt, email>,
reporter: IUser<_id, name, uname>[],
description: Array<string>,
moderationInfo: ...
}
``` 

Why the above recommendation:

The user reports are amazingly different from user-generated content reporting, especially we remove the `messages` variable; that was the main reason I did not proceed with the above schema for the message reports. So every time someone reports a user, we upsert the data. This covers the concern of constantly updating the data.

The above schema could have potential drawbacks, like making the queries based on `reporter` more complex. But at the same time, it simplifies the query that we use on the Main Console.
The main concern would be the increasing document size, as more reports against the user are submitted. However, I assume that won't be the case, considering we have a hard limit of 16MB per document. 

And please feel free to correct me if I miss assumed something or missed something. Thank you!


**Approach 2:**

It is good to have historical data for Moderation purposes. For example, if a user never had a `moderator` or any critical role, it helps the moderator understand the context better. And if the moderator sees that this user has changed over emails multiple times, that raises a red flag.

The `createdAt` is as well helpful information for Moderator; if a user created their account two days back and got multiple reports, it increases the probability that this account was created for malicious purposes.

_However, none of the above guarantees any fact; after all, the final action to take should be handled by the Moderators reasoning_

Thank you!

Reply:
Hey @Dnouv !
As I mentioned, we discussed these two proposals with the Architecture team and came to the conclusion that reports should behave like a snapshot of the moment of the report. Hence, we should store each of them in a separate record and do not upsert the reported user data (then, if someone is reported due to their username, we wouldn't lose track of it if the reported user updates it).
Still, inserting more data into the report sounds like a good idea. We now also store `emails` and the `createdAt` field.